clients/spring-boot-starter-camunda-sdk/src/test/java/io/camunda/spring/client/jobhandling/JobHandlerInvokingSpringBeansTest.java

```diff
@@ -0,0 +1,230 @@
+import io.camunda.spring.client.TestJobWorkers.JobResponse;
+import io.camunda.spring.client.TestJobWorkers.Response;
+import io.camunda.spring.client.TestJobWorkers.TestDimension;
+import io.camunda.spring.client.annotation.value.JobWorkerValue;
+import io.camunda.spring.client.jobhandling.parameter.DefaultParameterResolverStrategy;
+import io.camunda.spring.client.jobhandling.parameter.ParameterResolver;
+import io.camunda.spring.client.jobhandling.result.DefaultResultProcessorStrategy;
+import io.camunda.spring.client.jobhandling.result.ResultProcessor;
+import io.camunda.spring.client.metrics.DefaultNoopMetricsRecorder;
+import io.camunda.spring.client.metrics.MetricsRecorder;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class JobHandlerInvokingSpringBeansTest {
+
+  static Stream<Arguments> shouldThrowBpmnErrorSource() {
+    final List<Arguments> arguments = new ArrayList<>();
+    for (final AutoComplete autoComplete : AutoComplete.values()) {
+      for (final Response response : Response.values()) {
+        if (response.name().contains("BPMN_ERROR")) {
+          arguments.add(Arguments.of(autoComplete, response));
+        }
+      }
+    }
+    return arguments.stream();
+  }
+
+  static Stream<Arguments> shouldFailJobSource() {
+    final List<Arguments> arguments = new ArrayList<>();
+    for (final AutoComplete autoComplete : AutoComplete.values()) {
+      for (final Response response : Response.values()) {
+        if (response.name().contains("JOB_ERROR")) {
+          arguments.add(Arguments.of(autoComplete, response));
+        }
+      }
+    }
+    return arguments.stream();
+  }
+
+  private static CommandExceptionHandlingStrategy commandExceptionHandlingStrategy() {
+    return new DefaultCommandExceptionHandlingStrategy(
+        BackoffSupplier.newBackoffBuilder().build(),
+        CamundaClientExecutorService.createDefault().get());
+  }
+
+  private static MetricsRecorder metricsRecorder() {
+    return new DefaultNoopMetricsRecorder();
+  }
+
+  private static List<ParameterResolver> parameterResolvers(final JobWorkerValue jobWorkerValue) {
+    return JobHandlingUtil.createParameterResolvers(
+        new DefaultParameterResolverStrategy(new CamundaObjectMapper()), jobWorkerValue);
+  }
+
+  private static ResultProcessor resultProcessor(final JobWorkerValue jobWorkerValue) {
+    return JobHandlingUtil.createResultProcessor(
+        new DefaultResultProcessorStrategy(), jobWorkerValue);
+  }
+
+  private static JobExceptionHandlingStrategy jobExceptionHandlingStrategy() {
+    return new DefaultJobExceptionHandlingStrategy();
+  }
+
+  @ParameterizedTest
```

Comment:
💭  nit-pick opinion on style, let's move the test methods to the top and helpers last, I would usually be interested in the tests first opening a test class

Reply:
Agree